### PR TITLE
Add the networked resourceLink and the associated urlHelpers and test

### PR DESCRIPTION
**Pull request by epeach**
NetworkResourceLink is a wrapper around ResourceLink that uses the reference link in the props to get a text description for the ResourceLink. For this it uses a urlHelper function and memoizes it to minimize re-requests. The behavior is that the ResourceLink is displayed with its url until the urlHelper fetch returns. If the reference link that is passed to the urlHelper does not exists or cannot be parsed as we expect, we return the reference link and display just that.

### CODE REVIEW

File: apps/src/lib/util/urlHelpers.js [hunk 1]
```diff
@@ -13,3 +15,37 @@ export function pegasus(relativeUrl) {
...
   return relativeUrl;
 }
+
+/**
+ * Retrieve the meta description tag from the specified url
+ * Memoize so that we only request once per relative url.
+ */
+export function metaTagDescription(relativeUrl) {
+  return metaTagMemoized(relativeUrl);
+}
```
Comment (islemaster) [inline apps/src/lib/util/urlHelpers.js]: I'm pretty sure this pass-through is unnecessary...
```js
export const metaTagDescription = _.memoize((relativeUrl) => {
```

File: apps/src/lib/util/urlHelpers.js [hunk 2]
```diff
@@ -13,3 +15,37 @@ export function pegasus(relativeUrl) {
...
+            if (metaTag) {
+              return metaTag;
+            } else {
+              return relativeUrl;
+            }
+          });
+        }
+      }
+    ).catch(error => relativeUrl);
+});
```
Comment (islemaster) [inline apps/src/lib/util/urlHelpers.js]: I'm convinced this works great, but there are some style things, especially with promise chains, that could be cleaned up.

In general, promise chains inside promise chains is a code smell, and might indicate you should break the work up differently.  Here's what I'd do:

```js
return fetch(relativeUrl)
  .then(response => Promise.all([response.status, response.text()]))
  .then((status, text) => {
    if (status < 200 || status >= 300) {
      return relativeUrl;
    } else {
      const metaTag = $(text).filter("meta[name='description']").attr('content');
      return metaTag || relativeUrl;
    }
  })
  .catch(() => relativeUrl);
```
The first `.then` clause does a couple of things - it does additional async work, and it only passes along the parts of the response we care about.  This is slightly less optimal than your solution (you're only doing the async work of reading the response body if the status is OK) but I think it's easier to follow.

File: apps/src/templates/instructions/NetworkResourceLink.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    icon: PropTypes.string.isRequired,
+    reference: PropTypes.string.isRequired,
+  };
+
+  state = {
+    title: null
+  };
+
+  componentDidMount() {
+    metaTagDescription(this.props.reference).then(title => this.setState({title}));
```
Comment (islemaster) [inline apps/src/templates/instructions/NetworkResourceLink.js:17]: Beautiful!

File: apps/src/lib/util/urlHelpers.js (hunk 2 above)
Comment (epeach) [reply apps/src/lib/util/urlHelpers.js]: This is beautiful! I knew there had to be a way to make this promise chain readable and prettier. Thanks for the assist!

**Pull request merged**